QuadradiusR/scripts/torus.gd

```diff
@@ -10,33 +10,49 @@ onready var light_on = $LightGroup/LightOn
 onready var light_off = $LightGroup/LightOff
 onready var shadow = $TorusShadow
 onready var collision_detector = $MouseDetector/CollisionShape2D
+onready var powers_list = $PowerupPopup
 
 const starting_pos = Vector2(50, 50)
 const color_asset_path = "res://original_assets/game/sprites/{color}/{mode}.png"
-const color_textures = {COLORS.RED: "DefineSprite_412_Decoration0",
-						COLORS.BLUE: "DefineSprite_415_Decoration1",
-						COLORS.GREEN: "DefineSprite_418_Decoration2",
-						COLORS.YELLOW: "DefineSprite_421_Decoration3",
-						COLORS.CYAN: "DefineSprite_856_Decoration4",
-						COLORS.MAGENTA: "DefineSprite_859_Decoration5"}
+const color_textures = {
+	COLORS.RED: "DefineSprite_412_Decoration0",
+	COLORS.BLUE: "DefineSprite_415_Decoration1",
+	COLORS.GREEN: "DefineSprite_418_Decoration2",
+	COLORS.YELLOW: "DefineSprite_421_Decoration3",
+	COLORS.CYAN: "DefineSprite_856_Decoration4",
+	COLORS.MAGENTA: "DefineSprite_859_Decoration5"
+}
+const color_tint_codes = {
+	COLORS.RED: Color("ffd7d7"),  # all of those values are eyeballed, correct if needed
+	COLORS.BLUE: Color("b2c5ff"),
+	COLORS.GREEN: Color("aeffbe"),
+	COLORS.YELLOW: Color("fbffae"),
+	COLORS.CYAN: Color("aefff7"),
+	COLORS.MAGENTA: Color("ffaefa")
+}
 
 var movement_powerup_manager = preload("res://scripts/movement_powerup_manager.gd").new()
 var board: Control
 var piece_id: String
 var player_id: String
+var powerup_id_to_definition_id = {}  # all powerups owned by player with their types
+var powerup_definition_id_to_count = {}  # shorthand for count of powerups by type
 var color = COLORS.RED
-var current_tile: Tile
-var is_held = false
-var can_interact = true
-var powerup_id_to_definition_id = {}
+var current_tile: Tile  # the tile on which the piece stands
+var is_held = false  # is player currently dragging this piece
+var can_interact = true  # can player move the piece
+var powers_list_open = false  # is the popup with powers open
+var selected = false  # has player clicked the piece
 
 
 
+# # # # #
+# Init  #
+# # # # #
 func _ready():
 	set_process(false)
-	light_off.texture = load(color_asset_path.format({"color": color_textures[color], "mode": 1}))
-	light_on.texture = load(color_asset_path.format({"color": color_textures[color], "mode": 2}))
-	update_shadow(get_parent().get_parent().get_parent().tile_pos)
+	_set_color(self.color)
```

Comment:
this sets up the color used by the piece, is dependent on the view on the piece (owner and opponent may see them in different colors)